libimage/filters.go

```diff
@@ -0,0 +1,205 @@
+package libimage
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	filtersPkg "github.com/containers/common/pkg/filters"
+	"github.com/pkg/errors"
+)
+
+// filterFunc is a prototype for a positive image filter.  Returning `true`
+// indicates that the image matches the criteria.
+type filterFunc func(*Image) (bool, error)
```

Comment:
Do you have an example of a filterfunc that returns an error?  Should we just return a bool?

Reply:
`filterLabel` or `filterIntermediate` return errors.